### ISSUE TITLE
Initial audit of header hierarchy for all module API pages

### ISSUE THREAD

**Issue by Eric-Arellano** [Qiskit/documentation#942]
Like https://github.com/Qiskit/qiskit/pull/11923, but for all the other modules. We want to look for a  few things:

* We don't jump from h1 to h3, which is an accessibility violation. When this happens, we should an h2 to the API doc
* There aren't any unintended heading mistakes like how in https://github.com/Qiskit/qiskit/pull/11923 we were making something a lower header level without realizing.

Note that we should never use more than one h1 on module pages; don't try to improve the header hierarchy that way.

Module pages can be identified by searching for `python_api_type: module`. This shouldn't be a problem on function and class API pages, which are much more auto-generated than human-written.

--

- [x] qiskit.assembler
- [x] qiskit.circuit
- [x] qiskit.circuit.classical
- [x] qiskit.circuit.classicalfunction
- [x] qiskit.circuit.library
- [x] qiskit.circuit.singleton
- [x] qiskit.compiler
- [x] qiskit.converters
- [x] qiskit.dagcircuit
- [x] qiskit.exceptions
- [x] qiskit.passmanager
- [x] qiskit.primitives
- [x] qiskit.providers
- [x] qiskit.providers.basic_provider
- [x] qiskit.providers.fake_provider
- [x] qiskit.providers.models
- [x] qiskit.pulse
- [x] qiskit.qasm2
- [x] qiskit.qasm3
- [x] qiskit.qobj
- [x] qiskit.qpy
- [x] qiskit.quantum_info
- [x] qiskit.result
- [x] qiskit.scheduler
- [x] qiskit.synthesis
- [x] qiskit.synthesis.unitary.aqc
- [x] qiskit.transpiler
- [x] qiskit.transpiler.passes
- [x] qiskit.transpiler.passes.synthesis.plugin
- [x] qiskit.transpiler.preset_passmanagers
- [x] qiskit.transpiler.preset_passmanagers.plugin
- [x] qiskit.utils
- [x] qiskit.visualization
